dnaThreeBit/convert.go

```diff
@@ -0,0 +1,89 @@
+package dnaThreeBit
+
+import (
+	"bytes"
+	"github.com/vertgenlab/gonomics/dna"
+	"log"
+)
+
+func RuneToThreeBitBase(r rune) ThreeBitBase {
+	switch r {
+	case 'A':
+		return A
+	case 'C':
+		return C
+	case 'G':
+		return G
+	case 'T':
+		return T
+	case 'N':
+		return N
+	case 'a':
+		return A
+	case 'c':
+		return C
+	case 'g':
+		return G
+	case 't':
+		return T
+	case 'n':
+		return N
+	default:
+		log.Fatalf("Error: unexpected character in dna %c\n", r)
+		return N
+	}
+}
+
+func ThreeBitBaseToRune(base ThreeBitBase) rune {
+	switch base {
+	case A:
+		return 'A'
+	case C:
+		return 'C'
+	case G:
+		return 'G'
+	case T:
+		return 'T'
+	case N:
+		return 'N'
+	default:
+		log.Fatalf("Error: unexpected value in dna Base when converting to rune\n")
+		return 'N'
+	}
+}
+
+func ThreeBitBaseToString(b ThreeBitBase) string {
+	return string(ThreeBitBaseToRune(b))
+}
+
+func FromString(s string) *ThreeBit {
+	answer := &ThreeBit{Seq: []uint64{}, Len: 0}
+	for _, runeValue := range s {
+		answer = Append(answer, RuneToThreeBitBase(runeValue))
+	}
+	return answer
+}
+
+func ToString(fragment *ThreeBit) string {
+	var buffer bytes.Buffer
```

Comment:
something I learned recently is after go 1.13, the strings library has something similar to bytes.Buffer but optimized for concatenating strings changing this is a decent speed up. strings.Builder comes out ahead in situations like this one.  I can send you some of the benchmarks I did over slack.

```
func ToString(fragment *ThreeBit) string {
	var buffer strings.Builder

	for i := 0; i < fragment.Len; i++ {
		buffer.WriteRune(dna.BaseToRune(GetBase(fragment, i)))
	}
	return buffer.String()
}
```